src/ol/layer/Graticule.js

```diff
@@ -479,11 +479,25 @@ class Graticule extends VectorLayer {
    * @return {Array<import("../extent.js").Extent>} Extents.
    */
   strategyFunction(extent, resolution) {
-    if (this.loadedExtent_ && !equals(this.loadedExtent_, extent)) {
+    // extents may be passed in different worlds, to avoid endless loop we use only one
+    const realExtent = extent.slice();
+    if (this.projection_) {
+      const center = getCenter(extent);
+      const projectionExtent = this.projection_.getExtent();
+      const worldWidth = getWidth(projectionExtent);
+      if (this.getSource().getWrapX() && this.projection_.canWrapX() && !containsExtent(projectionExtent, extent)) {
+        const worldsAway = Math.floor((center[0] - projectionExtent[0]) / worldWidth);
+        realExtent[0] -= (worldsAway * worldWidth);
```

Comment:
We're doing this same operation a lot. I think a `toRealWorld(extent, projection)` function in `ol/extent` could be nice.

Reply:
Once in the renderer and once in the graticule.  Is it being done anywhere else?

Reply:
I added another commit that adds wrapX functions for extent and coordinate. With that commit, I also simplified the solution your rounding problem like I suggested.

Can you please take another final look?

Reply:
The wrapX functions needed to allow `worldsAway` to be calculated when the center is at +180 otherwise the graticule loops when the view is centered on the dateline.  The strategy must test for `!equal` as any real extent change needs the graticule to be updated.  With `!containsExtent` the graticule intervals do not update when zooming it.  I cannot see any alternative to rounding as the problem is similar to JavaScript 0.2 + 0.1 and not related to resolution.

Reply:
I know that it is not *related* to resolution, but *using* resolution for rounding is nicer than arbitrary numbers. In the end what you need is pixel exact extents, and that can be achieved with rounding by resolution (i.e. 1 pixel).